File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    ans = 0
    
    for i in range(len(moves)):
        grab = 0
        x = moves[i] - 1
        
        for j in range(len(board)):
            if board[j][x] != 0:
                grab = board[j][x]
                board[j][x] = 0
        
                if not stack:
                    stack.append(grab)
                elif stack[-1] == grab:
                        stack.pop()
                        ans += 2
                else:
                    stack.append(grab)
                
                break
                    
                    
    return ans

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> stack = new Stack<Integer>();
        int length = board[0].length;
        int answer = 0;
        
        for (int i = 0; i < moves.length; i++) {
            int now = moves[i] - 1;
            
            if (stack.isEmpty()) {
                for (int j = 0; j < length; j++) {
                    if (board[j][now] != 0) {
                        stack.push(board[j][now]);
                        board[j][now] = 0;
                        break;
                    }
                }
            } else {
                for (int j = 0; j < length; j++) {
                    if (board[j][now] != 0) {
                        if (board[j][now] == stack.lastElement()) {
                            answer+=2;
                            stack.pop();
                            board[j][now] = 0;
                        } else {
                            stack.push(board[j][now]);
                            board[j][now] = 0;
                        }
                        
                        break;
                    }
                }
            }
        }
        return answer;
    }
}